### ISSUE TITLE
Windows image: Error uploading ps script containing env vars, Couldn't create shell

### ISSUE THREAD

**Issue by marcosmamg** [hashicorp/packer#7956]
Hello Team, I am running into this issue:

`googlecompute: Error processing command: Error uploading ps script containing env vars: Error uploading file to $env:TEMP\winrmcp-432e2c29-dea5-4d3f-56ce-44f4b2a98a47.tmp: Couldn't create shell: http response error: 401 - invalid content type
`

My base image is this hardened windows server image: 
https://console.cloud.google.com/marketplace/details/cis-public/cis-windows-server-2016-level-1

I am running a sysprep script which is actually the one ansible uses to prepare instances to connect through winrm, but I am enabling only HTTPS. I am not using basic auth instead I am using NTLM with SSL.
https://github.com/ansible/ansible/blob/devel/examples/scripts/ConfigureRemotingForAnsible.ps1

Packer connects correctly to the instance, but it appears that it cannot upload this ps1 file of the provisioners and then it hangs and does not execute the provisioners.

**Packer version:** 1.4.1
**Host platform:** MacOs Mojave 10.14.5
**Debug log output from `PACKER_LOG=1:** https://gist.github.com/marcosmamg/8e46f7663ce571ba9c27ccda3d7ef1f3
**Template.json:**
https://gist.github.com/marcosmamg/add9636aed2a9c3a644f82b5a32394b3

**Comment by marcosmamg** [comment]
By getting into the instance I just realized that this registry entry `LocalAccountTokenFilterPolicy` it not set to 1, therefore packer cannot connect and it appears there is an issue with evelation or running commands as admin. If I get into the instance and I manually change it, packer is able to complete everything with no issues.

But the thing is that this policy should be updated with the sysprep script I am using:
https://github.com/ansible/ansible/blob/devel/examples/scripts/ConfigureRemotingForAnsible.ps1#L297

**Comment by SwampDragons** [comment]
It looks like you enable WinRM before you fix the registry setting. What happens if you move the part of the code that configures WinRM to the very end of your sysprep script, so that Packer is unable to connect until after the registry edit has been applied?

**Comment by SwampDragons** [comment]
(note that this doesn't make a ton of sense to me because of the 10 minute pause you've added, but it's worth ruling out)

Does the system require a restart for the registry key to be applied?

**Comment by marcosmamg** [comment]
I did some research and this is due to a local group policy in windows for this specific image,
even though I updated the registry key the policy overrides it.

What needs to be done is that you must update the local group policy instead.
The policy is called: Apply UAC restrictions to local accounts on network logons from the group: 
SCM: Pass the Hash Mitigations group policy settings.

I followed this example to be able to update the GPO for this non-domain joined instance.
https://www.rootusers.com/deploy-configurations-to-domain-and-non-domain-joined-servers-with-security-compliance-manager-scm/

**Comment by sjauld** [comment]
@marcosmamg do you have a powershell snippet that will update this policy? I seem to be suffering from the same issue using the AWS CIS Win 2016 image 😢

**Comment by marcosmamg** [comment]
To solve the group policy issue with CIS image:
- Install SCM (https://www.microsoft.com/en-us/download/details.aspx?id=53353)
- Download the latest Baselines for Windows Server 2016
- Duplicate the baseline WS2016 Member Security Compliance
- Delete all of the other settings and leave only: "Apply UAC restrictions to local accounts on network logons". I did this to prevent any undesired change.
- Set the Disabled value to the policy and export it as GPO Backup Folder
- Search for the registry.pol file in the backup folder and import it using LGPO.EXE (https://blogs.technet.microsoft.com/secguide/2016/01/21/lgpo-exe-local-group-policy-object-utility-v1-0/)
- .\LGPO.exe /m registry.pol
- Voila, the Registry Key (HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\policies\system\LocalAccountTokenFilterPolicy) is now set to 1 or disabled and everything works.

Note: I tried to change the registry key with Powershell in the sysprep scripts, but I did not get it to work, and my understanding is because GPOs are supposed to override the values and make them persistent.

Give it a try!!!
